File: stock-market-service/src/main/java/rs/edu/raf/mapper/ListingOwnerMapper.java
package rs.edu.raf.mapper;

import org.springframework.stereotype.Component;
import rs.edu.raf.dto.ListingOwnerDTO;
import rs.edu.raf.model.ListingOwner;

@Component
public class ListingOwnerMapper {

    public ListingOwnerDTO listingOwnerToListingOwnerDTO(ListingOwner listingOwner) {
        return new ListingOwnerDTO(listingOwner.getId(), listingOwner.getOwner(), listingOwner.getTicker(),
                listingOwner.getQuantity());
    }
}

File: stock-market-service-kotlin/src/main/kotlin/org/example/stockmarketservicekotlin/mapper/ListingOwnerMapper.kt
package org.example.stockmarketservicekotlin.mapper

import org.example.stockmarketservicekotlin.dto.ListingOwnerDTO
import org.example.stockmarketservicekotlin.model.ListingOwner
import org.springframework.stereotype.Component

@Component
class ListingOwnerMapper {
    fun listingOwnerToListingOwnerDTO(listingOwner: ListingOwner): ListingOwnerDTO {
        return ListingOwnerDTO(
            listingOwner.id, listingOwner.owner, listingOwner.ticker, listingOwner.quantity
        )
    }
}